### DISCUSSION TITLE
Motors not responding to GCODE and Homing Cycle, and jogging way too slow

### DISCUSSION THREAD

**Discussion by Godinspector** [winder/Universal-G-Code-Sender#1770] [Help]
**Describe the bug**
Hello, i'm having a problem wich i don't really know how to fix. I have just built my first CNC Machine it seems to be everything fine with wiring and all. i can jog my motors : X+, X-, Y+, Y-, Z+ and Z- all working fine (but they're moving way too slow).  when i  click in "open" the alarm comes up. I type $H and the machine seems to do a little movement in Z axis and then the alarm comes again. No X or Y homing movements. The alarm only desappears when i type $X.

   The message when i click $H:

">>> $H
[Error] An error was detected while sending '$H': ALARM: Homing fail. Streaming has been paused.
[Error] Error while processing response <ok>: An unexpected command was completed by the controller.
Grbl 0.9j ['$' for help]
>>> $$
['$H'|'$X' to unlock]
$0 = 10    (step pulse, usec)
$1 = 25    (step idle delay, msec)
$2 = 0    (step port invert mask:00000000)
$3 = 0    (dir port invert mask:00000000)
$4 = 0    (step enable invert, bool)
$5 = 1    (limit pins invert, bool)
$6 = 0    (probe pin invert, bool)
$10 = 3    (status report mask:00000011)
$11 = 0.010    (junction deviation, mm)
$12 = 0.002    (arc tolerance, mm)
$13 = 0    (report inches, bool)
$20 = 0    (soft limits, bool)
$21 = 1    (hard limits, bool)
$22 = 1    (homing cycle, bool)
$23 = 3    (homing dir invert mask:00000011)
$24 = 60.000    (homing feed, mm/min)
$25 = 700.000    (homing seek, mm/min)
$26 = 250    (homing debounce, msec)
$27 = 1.000    (homing pull-off, mm)
$100 = 800.000    (x, step/mm)
$101 = 800.000    (y, step/mm)
$102 = 800.000    (z, step/mm)
$110 = 700.000    (x max rate, mm/min)
$111 = 700.000    (y max rate, mm/min)
$112 = 700.000    (z max rate, mm/min)
$120 = 50.000    (x accel, mm/sec^2)
$121 = 50.000    (y accel, mm/sec^2)
$122 = 50.000    (z accel, mm/sec^2)
$130 = 200.000    (x max travel, mm)
$131 = 200.000    (y max travel, mm)
$132 = 40.000    (z max travel, mm)
ok
>>> $G
[G0 G54 G17 G21 G90 G94 M0 M5 M9 T0 F0.]
ok"

I tryied to run a simple Gcode and the motors just started making a big noise, and not moving at all

 I am using the UGS 2.0.7 with GRBL 0.9J
My operating system is Windows10

![Captura de Tela (341)](https://user-images.githubusercontent.com/89892126/131613673-71231526-89ad-4235-b46d-7a125e17ecd9.png)
![Captura de Tela (342)](https://user-images.githubusercontent.com/89892126/131613681-f3bfd0f0-b150-4076-919f-8ac027132195.png)
![Captura de Tela (343)](https://user-images.githubusercontent.com/89892126/131613683-8ab37216-f3f3-4b43-9055-0a49a7fe2609.png)
 

Hope that you guys can help me

**Comment by breiler** [comment]
I feel bad for commenting on this so late, probably not relevant anymore... =/

* Your jog feed rate is set to 25 mm/minute which is pretty slow, increase that in the jog settings.
* Before you setup homing you should make sure your step settings are correct $100, $101, $102 so that when you jog 1mm the machine is actually moving 1mm.
* How large is your work area of the machine? The $130-$132 setting specify how large your work area is and is used for how far the machine should seek for the limit switch. In your case 200 x 200 x 200mm. If this is set to low it will not find the limit switch and throw the alarm you mentioned `An error was detected while sending '$H': ALARM: Homing fail.`
